server/data/data-jmap/src/main/java/org/apache/james/jmap/api/filtering/Rule.java

```diff
@@ -371,6 +369,10 @@ public Builder setForward(Optional<Forward> forward) {
             }
 
             public Action build() {
+                Preconditions.checkNotNull(appendInMailboxes, "appendInMailboxes should not be null");
+                Preconditions.checkNotNull(withKeywords, "withKeywords should not be null");
+                Preconditions.checkNotNull(forward, "forward should not be null");
```

Comment:
Not sure against those TBH

Reply:
could you give more details?

Reply:
as first version of actions if I remember correctly only had `appendInMailboxes`, so that could break histories with those

Reply:
```
public static Action of(AppendInMailboxes appendInMailboxes) {
            return new Action(appendInMailboxes, false, false, false, ImmutableList.of(), Optional.empty());
        }
```
if we do not add withKeywords, it will be empty list by default. I have tried with null value for withKeywords before and NPE occur

Reply:
>  I have tried with null value for withKeywords before and NPE occurred

Wrap it with Optional and handle the null? I do not think always assuming it not null is right.

Reply:
Might have mixed with something else, ok